bind.go

```diff
@@ -35,32 +35,70 @@ func Named(name string, value interface{}) driver.NamedValue {
 	}
 }
 
+var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+var bindPositionalRe = regexp.MustCompile(`[?]`)
+
 func bind(tz *time.Location, query string, args ...interface{}) (string, error) {
 	if len(args) == 0 {
 		return query, nil
 	}
 	var (
-		haveNamed   bool
-		haveNumeric bool
+		haveNamed      bool
+		haveNumeric    bool
+		havePositional bool
 	)
+	haveNumeric = bindNumericRe.MatchString(query)
+	havePositional = bindPositionalRe.MatchString(query)
+	if haveNumeric && havePositional {
+		return "", ErrBindMixedParamsFormats
+	}
 	for _, v := range args {
 		switch v.(type) {
 		case driver.NamedValue:
 			haveNamed = true
 		default:
-			haveNumeric = true
 		}
-		if haveNamed && haveNumeric {
-			return "", ErrBindMixedNamedAndNumericParams
+		if haveNamed && (haveNumeric || havePositional) {
+			return "", ErrBindMixedParamsFormats
 		}
 	}
 	if haveNamed {
 		return bindNamed(tz, query, args...)
 	}
-	return bindNumeric(tz, query, args...)
+	if haveNumeric {
+		return bindNumeric(tz, query, args...)
+	}
+	return bindPositional(tz, query, args...)
 }
 
-var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+func bindPositional(tz *time.Location, query string, args ...interface{}) (_ string, err error) {
+	var (
```

Comment:
Can this be simplified just by using a param array?  Since we don't have string keys just indexes seem sufficient.